### ISSUE TITLE
filters.median_blur: an even kernel_size raises a raw reshape error instead of being rejected

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4781]
`kornia.filters.median_blur` (and `RandomMedianBlur`, which delegates to it) accepts an even `kernel_size` without validation and then fails in its own `view` with a raw shape error that names neither the kernel nor the odd-size requirement. Below the kernel size the same call fails one layer lower, inside `F.conv2d`.

## Repro

```python
import torch, kornia, kornia.augmentation as K

for shape in [(1, 1, 8, 8), (1, 1, 1, 1)]:
    for ks in [(4, 4), (3, 4)]:
        try:
            kornia.filters.median_blur(torch.rand(*shape), ks)
        except Exception as e:
            print(shape, ks, type(e).__name__, e)
torch.manual_seed(0)
K.RandomMedianBlur((4, 4), p=1.0)  # constructs fine
try:
    K.RandomMedianBlur((4, 4), p=1.0)(torch.rand(2, 3, 32, 32))
except Exception as e:
    print("RandomMedianBlur", type(e).__name__, e)
```

## Observed

```
(1, 1, 8, 8) (4, 4) RuntimeError shape '[1, 1, 16, 8, 8]' is invalid for input of size 784
(1, 1, 8, 8) (3, 4) RuntimeError shape '[1, 1, 12, 8, 8]' is invalid for input of size 672
(1, 1, 1, 1) (4, 4) RuntimeError Calculated padded input size per channel: (3 x 3). Kernel size: (4 x 4). Kernel size can't be greater than actual input size
RandomMedianBlur RuntimeError shape '[2, 3, 16, 32, 32]' is invalid for input of size 92256
```

## Expected

An even entry has no centred median window, and the zero padding `(k - 1) // 2` makes the output one pixel smaller than the input per even axis, so the `view` back to `(B, C, kH * kW, H, W)` can never succeed. Either:

1. `median_blur` rejects an even entry with a `KORNIA_CHECK` naming the kernel (as `gaussian_blur2d` does with its "odd integer" error), ideally also at `RandomMedianBlur` construction; or
2. an even window is supported with an asymmetric pad, as `box_blur` does.

## Why it is a bug

It is a precondition kornia never validates; the failure is torch's reshape error from inside kornia, not a message about the argument the caller passed. `RandomGaussianBlur` rejects the same input with a named error. PR #4673 (border types for `median_blur`) keeps this behaviour.

## Environment

kornia `main` @ 57cd68e6f, torch 2.14.0, python 3.11, macOS arm64 cpu.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
